corehq/form_processor/backends/sql/dbaccessors.py

```diff
@@ -248,6 +249,35 @@ def _get_forms_for_user(domain, user_id, state, ids_only=False):
             return [form.form_id for form in forms]
         return forms
 
+    @staticmethod
+    def get_all_forms_received_since(received_on_since=None, chunk_size=500):
+        start_from = received_on_since or datetime.min
+        # todo: this will greedily query the same cases multiple times in a sharded
+        # setup. We should make it smarter
+        batch = FormAccessorSQL.get_forms_received_since(start_from, limit=chunk_size)
+        while len(batch) == chunk_size:
```

Comment:
could this just be `while len(batch) > 0` and then we don't have the `# last batch` logic?